storm/src/main/storm/mesos/schedulers/SchedulerUtils.java

```diff
@@ -31,49 +34,73 @@
 import java.util.List;
 import java.util.Map;
 
+import static storm.mesos.resources.ResourceEntries.RangeResourceEntry;
+import static storm.mesos.resources.ResourceEntries.ScalarResourceEntry;
+
 public class SchedulerUtils {
 
   private static final Logger log = LoggerFactory.getLogger(SchedulerUtils.class);
 
-  public static boolean isFit(Map mesosStormConf, OfferResources offerResources, TopologyDetails topologyDetails, boolean supervisorExists) {
+  public static List<RangeResourceEntry> getPorts(OfferResources offerResources, int requiredCount) {
+    List<RangeResourceEntry> retVal = new ArrayList<>();
+    List<RangeResourceEntry> resourceEntryList = offerResources.getAllAvailableResources(ResourceType.PORTS);
+
+    for (RangeResourceEntry rangeResourceEntry : resourceEntryList) {
+      Long begin = rangeResourceEntry.getBegin();
+      Long end = rangeResourceEntry.getEnd();
+      for (int i = 0; i <= (end - begin) && requiredCount > 0; i++) {
```

Comment:
This is touching the core logic that is actively being used. Will come back to this later. TODO(ksoundararaj).